explorer/testdata/class/fail_abstract_method_base_instance.carbon

```diff
@@ -0,0 +1,23 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// RUN: %{not} %{explorer-run}
+// RUN: %{not} %{explorer-run-trace}
+
+package ExplorerTest api;
+
+abstract class A {
+    abstract fn Foo[self: Self]();
+}
+
+base class B extends A {
```

Comment:
Would it be better to have a compile-time error here saying `Foo` is not implemented in a non-abstract class. If we can diagnose something statically, we definitely should.

Reply:
Following the table in https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/classes.md#inheritance, it is legal to have an `abstract` methods not mentioned in a derived class that is not final (i.e. `base` or `abstract`). But such a class is illegal to instantiate, hence the "type check" validation.

This is done during type-checking, to the closest to what "compile-time" could be for Explorer